marconi-sidechain/src/Marconi/Sidechain/Api/Routes.hs

```diff
@@ -1,40 +1,132 @@
 -- | Defines REST and JSON-RPC routes
 
-{-# LANGUAGE DataKinds     #-}
-{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeOperators     #-}
 
 module Marconi.Sidechain.Api.Routes where
 
+import Cardano.Api qualified as C
+import Cardano.Api.Shelley qualified as C
+import Data.Aeson (FromJSON (parseJSON), ToJSON (toJSON), Value (Object), object, (.:), (.=))
 import Data.Text (Text)
-import Marconi.Sidechain.Api.Types (UtxoQueryResult)
+import Data.Word (Word64)
+import GHC.Generics (Generic)
+import Marconi.ChainIndex.Indexers.MintBurn qualified as MpsTx
+import Marconi.ChainIndex.Indexers.Utxo qualified as Utxo
 import Network.JsonRpc.Types (JsonRpc, RawJsonRpc)
 import Servant.API (Get, JSON, PlainText, (:<|>), (:>))
 
-------------------------------------------
-  --  RPC types
-  --  methodName -> parameter(s) -> return-type
-------------------------------------------
-type RpcEcho = JsonRpc "echo" String String String
-
-type RpcUtxoQueryResult = JsonRpc "getUtxoFromAddress" String String UtxoQueryResult
-
-type RpcTargetAddresses = JsonRpc "getTargetAddresses" String String [Text]
+-- | marconi-sidechain APIs
+type API = JsonRpcAPI :<|> RestAPI
 
--- | Rpc routes
-type RpcAPI = RpcEcho :<|> RpcTargetAddresses :<|> RpcUtxoQueryResult
+----------------------------------------------
+--  RPC types
+--  methodName -> parameter(s) -> return-type
+----------------------------------------------
 
 -- | JSON-RPC API, endpoint
 type JsonRpcAPI = "json-rpc" :> RawJsonRpc RpcAPI
 
+-- | RPC routes
+type RpcAPI = RpcEchoEndpoint
+         :<|> RpcTargetAddressesEndpoint
```

Comment:
This is just a preference, and may not apply.
RPCs share the same edpoint, where to me, maybe incorrect, endpoint is is the location from which APIs can access the resources they need to carry out their function. And adding endpoint to the type name may mislead that each type has a different endpoint?

Reply:
Good point! So maybe use the `Method` suffix?